### PR TITLE
CC-28 Fix Mac OSX integration

### LINKED ISSUE #28: posgres container does not start in Mac OSX
@hedonistrh to fill the details

**One-line summary (optional)**

**Steps to Reproduce**
- Add screenshots to help explain your problem, if applicable
- Include specs (OS, Python version, code commit hash or release version), if useful

**Expected Behaviour**

**Design**

**Acceptance Criteria**

**Outcome**
- Clarify the deliverables (scripts, process etc.) and documentation, if applicable.

**Don't forget to**
- Point the ticket
- Set priority as a label
- Assign to a person
- Link dependencies, if applicable

### PR DESCRIPTION
**Pull request by sertansenturk**
## Description

closes #28 

## Motivation and Context

See #28 

## How Has This Been Tested?

- [x] Locally on Mac OSX by @sertansenturk 
- [x] Locally on Mac OSX by @hedonistrh
- [x] Locally on Ubuntu 16.04
- [x] on travis.ci

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] The code follows the code style of this project.
- [x] Tests for the changes have been added
- [x] Docs have been added / updated

### CODE REVIEW

**Comment by sertansenturk** [comment]
@hedonistrh As part of the review, can you:

- Ensure that the cookiecutter works by baking from the branch, i.e. `cookiecutter --checkout CC-28-postgres-does-not-start-in-mac-osx https://github.com/sertansenturk/cookiecutter-ds-docker`, and then run `make test`. Next, run `make`, verify that the stack runs fine, and you can log runs in mlflow.
- Go through "Files changed," check if you see any problems. Also if you have any parts you don't understand, please ask with inline comments. If I cannot explain, then there's a code smell...

File: CONTRIBUTORS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Halil Erdoğan	herdogan AT sabanciuniv DOT edu
```
Comment (sertansenturk) [inline CONTRIBUTORS]: @hedonistrh should we use another e-mail? :)

Comment (halil-erdogan) [reply CONTRIBUTORS]: We can use herdoganturkey@gmail.com 💯 :)

**Comment by halil-erdogan** [comment]
Hey @sertansenturk thanks a lot for your efforts and explanation. 

- [x] I have tried what you suggest and it works flawlessly. I have checked _ui_ of _Jupyter_ and _MLFlow_ and everything looks fine. Also I have stopped them after some works, there is no problem. 👍 

- [x] I also investigated _files changed_ and everything looks fine for me.

I have two questions
1. First one is about **WAIT_FOR_IT_TIMEOUT** Why we set that as 60? Like for slower processor, should we set that a bit higher to ensure everything starts fine? 
2. Did you write the script of wait for it or did you take it from elsewhere? If you got it from elsewhere, may you share it? _(Maybe you already mentioned it. If I missed it, sorry about that.)_

**Comment by sertansenturk** [comment]
Excellent questions. Let me explain:
>     1. First one is about **WAIT_FOR_IT_TIMEOUT** Why we set that as 60? 

Extending upon the second problem described in this [issue comment](https://github.com/sertansenturk/cookiecutter-ds-docker/issues/28#issuecomment-623674485): The main reason is the slow bootstrapping of the posgres container in Mac OSX. 

In my Mac, initializing the db in the first run takes around 30 seconds. If we don't wait for more (using `wait-for-it.sh`, which is explained in the next question), the `mlflow` container gives up waiting and the docker-compose stack fails. 60 seconds is overkill but a safe choice. Note that `wait-for-it.sh` waits at most 60 seconds, i.e. on my Ubuntu machine, `mlflow` container waits less than 3 seconds.

> Like for slower processor, should we set that a bit higher to ensure everything starts fine?

Potentially, but not sure if these containers should be run on less powerful machines. We might probably need to make other optimizations if such a need arises one day. Until then, this implementation meets its purpose...

>     2. Did you write the script of wait for it or did you take it from elsewhere? If you got it from elsewhere, may you share it? _(Maybe you already mentioned it. If I missed it, sorry about that.)_
It is a bash script taken from [this git repo](https://github.com/vishnubob/wait-for-it). The script is commonly used in docker-compose so that containers with dependencies wait properly. I took the origin as granted because it is famously referred in [docker-compose documentation](https://docs.docker.com/compose/startup-order/).

The script has an MIT license so we don't need to refer to the author, but I added the origin regardless for the sake of documentation. Thanks for pointing it out.

**Comment by halil-erdogan** [comment]
Thanks for the answer.

> 60 seconds is overkill but a safe choice .... Until then, this implementation meets its purpose...

So we can keep it like how it is currently. 💯 

> but I added the origin regardless for the sake of documentation

It is nice, thanks for doing that. This is also good for the future reference. :)

**Review by halil-erdogan — APPROVED** [review]
Changes to OSX integration works flawlessly. So, I am approving changes. 🥳

**Pull request merged**